sljit_src/sljitUtils.c

```diff
@@ -206,22 +206,34 @@ static SLJIT_INLINE sljit_s32 open_dev_zero(void)
 
 #if (defined SLJIT_UTIL_STACK && SLJIT_UTIL_STACK)
 
+#if !(defined SLJIT_UTIL_SIMPLE_STACK_ALLOCATION && SLJIT_UTIL_SIMPLE_STACK_ALLOCATION) 
 /* Planning to make it even more clever in the future. */
 static sljit_sw sljit_page_align = 0;
+#endif
 
 SLJIT_API_FUNC_ATTRIBUTE struct sljit_stack* SLJIT_FUNC sljit_allocate_stack(sljit_uw start_size, sljit_uw max_size, void *allocator_data)
 {
-	struct sljit_stack *stack;
-	void *ptr;
-#ifdef _WIN32
-	SYSTEM_INFO si;
-#endif
-
+	struct sljit_stack *stack; 
+	void *ptr; 
 	SLJIT_UNUSED_ARG(allocator_data);
 	if (start_size > max_size || start_size < 1)
+		return NULL; 
+#if (defined SLJIT_UTIL_SIMPLE_STACK_ALLOCATION && SLJIT_UTIL_SIMPLE_STACK_ALLOCATION) 
+	stack = (struct sljit_stack*)SLJIT_MALLOC(sizeof(struct sljit_stack), allocator_data);
+	if (!stack)
 		return NULL;
-
+	ptr = SLJIT_MALLOC(max_size, allocator_data);
+	if (ptr == NULL)
+	{
+		SLJIT_FREE(stack, allocator_data);
+		return NULL;
+	}
+	stack->min_start = (sljit_u8 *)ptr;
+ 	stack->end = stack->min_start + max_size;
+ 	stack->start = stack->end - start_size; 
+#else // SLJIT_UTIL_SIMPLE_STACK_ALLOCATION
 #ifdef _WIN32
+	SYSTEM_INFO si;
```

Comment:
Declaration after code is supported by msvc? Is it c99 compatible? I know some compilers complain about this, but I am not sure which standard introduced them.